### ISSUE TITLE
Assertion failed 'compIsaSupportedDebugOnly(InstructionSet_SSE2)'

### ISSUE THREAD

**Issue by BruceForstall** [dotnet/runtime#37260]
Over 52,000 tests in the `runtime-coreclr jitstress-isas-x86` pipeline failed (presumably all) with:
```
      Assert failure(PID 7268 [0x00001c64], Thread: 5492 [0x1574]): Assertion failed 'compIsaSupportedDebugOnly(InstructionSet_SSE2)' in 'VectorRelopTest`1[Single][System.Single]:VectorRelOp(float,float):int' during 'Importation' (IL size 1406)
      
          File: F:\workspace\_work\1\s\src\coreclr\src\jit\simdashwintrinsic.cpp Line: 870
          Image: C:\h\w\9D4B08E8\p\CoreRun.exe
```

Pipeline results:
https://dev.azure.com/dnceng/public/_build/results?buildId=665306&view=ms.vss-test-web.build-test-results-tab

Example:
https://helix.dot.net/api/2019-06-17/jobs/a36943e0-4275-4b16-9f01-6d656c33e5eb/workitems/JIT/console

```
    JIT\SIMD\VectorRelOp_r\VectorRelOp_r.cmd [FAIL]
      
      Assert failure(PID 7268 [0x00001c64], Thread: 5492 [0x1574]): Assertion failed 'compIsaSupportedDebugOnly(InstructionSet_SSE2)' in 'VectorRelopTest`1[Single][System.Single]:VectorRelOp(float,float):int' during 'Importation' (IL size 1406)
      
          File: F:\workspace\_work\1\s\src\coreclr\src\jit\simdashwintrinsic.cpp Line: 870
          Image: C:\h\w\9D4B08E8\p\CoreRun.exe
      
      
      Return code:      1
      Raw output file:      C:\h\w\9D4B08E8\w\A49A0919\e\JIT\SIMD\Reports\JIT.SIMD\VectorRelOp_r\VectorRelOp_r.output.txt
      Raw output:
      BEGIN EXECUTION
       "C:\h\w\9D4B08E8\p\corerun.exe" VectorRelOp_r.dll 
      Expected: 100
      Actual: -1073740286
      END EXECUTION - FAILED
      FAILED
      Test Harness Exitcode is : 1
      To run the test:
      > set CORE_ROOT=C:\h\w\9D4B08E8\p
      > C:\h\w\9D4B08E8\w\A49A0919\e\JIT\SIMD\VectorRelOp_r\VectorRelOp_r.cmd
      Expected: True
      Actual:   False
      Stack Trace:
        F:\workspace\_work\1\s\artifacts\tests\coreclr\Windows_NT.x86.Checked\TestWrappers\JIT.SIMD\JIT.SIMD.XUnitWrapper.cs(9646,0): at JIT_SIMD._VectorRelOp_r_VectorRelOp_r_._VectorRelOp_r_VectorRelOp_r_cmd()
      Output:
```

**Comment by BruceForstall** [comment]
@tannergooding @CarolEidt @echesakovMSFT

**Comment by tannergooding** [comment]
Will get a fix up, we are just missing a check to bail out if `COMPlus_EnableSSE2=0`

**Comment by VincentBu** [comment]
Hundreds of tests failed with `Assertion failed 'compIsaSupportedDebugOnly(InstructionSet_SSE2)'` again in [runtime-coreclr jitstress-isas-x86 20210522.1](https://dev.azure.com/dnceng/public/_build/results?buildId=1151759&view=ms.vss-test-web.build-test-results-tab&runId=34824852&resultId=110811&paneView=debug)

**One of Error messages:**
```

- JIT/SIMD/VectorInit_ro/VectorInit_ro.sh

Assert failure(PID 4765 [0x0000129d], Thread: 4765 [0x129d]): Assertion failed 'compIsaSupportedDebugOnly(InstructionSet_SSE2)' in 'VectorInitTest:VectorInit(float):int' during 'Morph - Global' (IL size 172)

File: /__w/1/s/src/coreclr/jit/gentree.cpp Line: 19356
Image: /datadisks/disk1/work/BD8A09C1/p/corerun
/datadisks/disk1/work/BD8A09C1/w/B53909D4/e/JIT/SIMD/VectorInit_ro/VectorInit_ro.sh: line 379:  4765 Aborted                 (core dumped) $LAUNCHER $ExePath "${CLRTestExecutionArguments[@]}"

Return code:      1
Raw output file:      /datadisks/disk1/work/BD8A09C1/w/B53909D4/e/JIT/SIMD/Reports/JIT.SIMD/VectorInit_ro/VectorInit_ro.output.txt
Raw output:
BEGIN EXECUTION
/datadisks/disk1/work/BD8A09C1/p/corerun VectorInit_ro.dll ''
Gathering state for process 4765 corerun
Writing minidump with heap to file /home/helixbot/dotnetbuild/dumps/coredump.4765.dmp
Written 52846592 bytes (12902 pages) to core file
Dump successfully written
Expected: 100
Actual: 134
END EXECUTION - FAILED
Test Harness Exitcode is : 1
To run the test:

set CORE_ROOT=/datadisks/disk1/work/BD8A09C1/p
/datadisks/disk1/work/BD8A09C1/w/B53909D4/e/JIT/SIMD/VectorInit_ro/VectorInit_ro.sh
Expected: True
Actual:   False


Stack trace
   at JIT_SIMD._VectorInit_ro_VectorInit_ro_._VectorInit_ro_VectorInit_ro_sh()
```

**Comment by tannergooding** [comment]
This has a fix pending review via #52985
